homeworks/pontazsolt/1_introduction/2_vars_datatypes/user_info.py

```diff
@@ -0,0 +1,74 @@
+import pprint
+
+
+#Original Dictionary
+user_info = {
+    "name": "Mike",
+    "age": 25,
+    "favourite_meals": [
+        "pizza",
+        "carbonara",
+        "sushi"
+    ],
+    "phone_contacts": {
+        "Mary": "+36701234567",
+        "Tim": "+36207654321",
+        "Tim2": "+36304567321",
+        "Jim": "+364005000"
+    }
+}
+
+
+
+#1. programming languages#
+programming_languages_correctable = input ("Write 4 programming language: ").strip()
+programming_languages = ",".join(programming_languages_correctable.split()) #szóközök eltávolítása, és szavak vesszővel elválasztása#
```

Comment:
Szia, szuper a megoldás, de bekérheted úgy is, hogy eleve vesszővel legyenek elválasztva, és akkor nincs plusz lépés.

Reply:
Erre nem találtam olyan megoldást ami ha esetleg vessző nélkül írja be a felhasználó szóközökkel, vagy esetleg több szóközzel akkor javítja automatikusan a kívánt formátumra, de én teljesen kezdő vagyok. Esetleg tudsz segíteni hogy is gondoltad?

Reply:
Szimplán csak arra gondoltam, hogy ami a mostani kódodban történik:

1. split()-tel felszeleteled a string-et a whitespace-ek mentén (tehát te fixen azt feltételezed hogy space-ekkel lesz elválasztva az adat), és így lesz egy listád.
2. Ezután ezt a listát a join()-nal újra stringgé alakítod, és ez a string lesz olyan amit már az elején is feltételezhetnél, hogy megkapod a felhasználótól. Erre gondoltam hogy mivel most komolyabb input validation-t úgysem csinálunk még, feltételezheted hogy vesszővel választja el a felhasználó ahogy kéred.
Ebben az esetben elég lenne csak annyi hogy:
`programming_languages_correctable = input ("Write 4 programming language: ").split(",")`

De mégegyszer mondom, nem hiba, sőt, tök jó hogy már a join()-t is ismered.